design-documents/testing/static/phpstan.md

```diff
@@ -0,0 +1,103 @@
+# Overview
+[PHPStan](https://github.com/phpstan/phpstan) is a code analysis tool which focuses on finding bugs without running the code.
+PHPStan uses [nikic/PHP-Parser](https://github.com/nikic/PHP-Parser) under the hood to build AST (abstract syntax tree) and perform sophisticated static code analysis.
+
+### Prerequisites
+- PHP >= 7.1.
+- Installed Magento application with generated code.
```

Comment:
Can we avoid installed instance? Is it only for generated code?

Reply:
Yes, we can. My bad, wrong wording. `setup:di:compile` is good enough.